scripts/generate-transcript.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+const moment = require("moment");
+const generateTranscript = require("discord-transcript-generator");
+const { TOKEN, ID, ISSUE_TITLE } = process.env;
+
+/**
+ * Extract a date string from an automatically generated TSC Meeting issue title.
+ * Issue title is formatted as "TSC meeting DD-MMMM-YYYY".
+ * https://github.com/eslint/eslint-github-bot/blob/master/src/plugins/recurring-issues/index.js#L202
+ * @param {string} issueTitle The issue title from which to extract the date string.
+ * @returns {string} The extracted date string.
+ */
+function extractDateString(issueTitle) {
+    const [, dateString = null] = /(\d\d-.+-\d\d\d\d)$/u.exec(issueTitle) || [];
+
+    if (!dateString) {
+        throw new Error("Can't extract date from issue title. Expecting format \"TSC meeting DD-MMMM-YYYY\".");
+    }
+
+    return dateString;
+}
+
+/**
+ * Formats a date string to MM/DD/YYYY.
+ * @param {string} dateString The date string to format.
+ * @returns {string} The formatted date string.
+ */
+function formatDate(dateString) {
+    return moment(new Date(dateString)).format("MM/DD/YYYY");
+}
+
+/**
+ * Generates the transcript file output path.
+ * @param {string} dateString The date string representing the date of the transcript.
+ * @returns {string} The path in which to output the transcript.
+ */
+function generateOutputPath(dateString) {
+    return `./notes/${moment(new Date(dateString)).format("YYYY-MM-DD")}-transcript.md`;
```

Comment:
This path should also include the intermediate year subdirectory.

The parse format would be `”MM/DD/YYYY”` if you’re interested in using moment’s parse format as above.

Reply:
Ah, good catch!